File: src/Program/ExtractFromDelims.cs
using System.Text.RegularExpressions;

namespace mcmli
{
    partial class Program
    {
        public static MatchCollection ExtractFromDelims(string text, string delims)
        {
            if (delims == "[]") return Regex.Matches(text, @"(?<=\[)([^]]*)(?=\])");
            if (delims == "()") return Regex.Matches(text, @"(?<=\()([^)]*)(?=\))");
            if (delims == "{}") return Regex.Matches(text, @"(?<=\{)([^}]*)(?=\})");

            return Regex.Matches(text,
                @"(?<=" + delims[0] + @")([^" + delims[1] + @"]*)(?=" + delims[1] + @")");
        }
    }
}

File: src/ExtractFromDelims.go
package main

import (
	"regexp"
)

// ExtractFromDelims comment
func ExtractFromDelims(Text, Delims string) []string {
	// re := regexp.MustCompile(`.*` + Delims[0] + `(.*?)` + Delims[1] + `.*`)
	re := regexp.MustCompile(`.*`)
	if Delims == "[]" {
		re = regexp.MustCompile(`\[(.*?)\]`)
	}
	if Delims == "<>" {
		re = regexp.MustCompile(`<(.*?)>`)
	}

	var Matches []string

	for _, match := range re.FindAllStringSubmatch(Text, -1) {
		Matches = append(Matches, match[1])
	}

	return Matches
}
